src/common/datasource/src/file_format/json.rs

```diff
@@ -203,4 +221,165 @@ mod tests {
             }
         );
     }
+
+    #[tokio::test]
+    async fn test_compressed_json() {
+        // Create test data
+        let column_schemas = vec![
+            ColumnSchema::new("id", ConcreteDataType::uint32_datatype(), false),
+            ColumnSchema::new("name", ConcreteDataType::string_datatype(), false),
+            ColumnSchema::new("value", ConcreteDataType::float64_datatype(), false),
+        ];
+        let schema = Arc::new(Schema::new(column_schemas));
+
+        // Create multiple record batches with different data
+        let batch1_columns: Vec<VectorRef> = vec![
+            Arc::new(UInt32Vector::from_slice(vec![1, 2, 3])),
+            Arc::new(StringVector::from(vec!["Alice", "Bob", "Charlie"])),
+            Arc::new(Float64Vector::from_slice(vec![10.5, 20.3, 30.7])),
+        ];
+        let batch1 = RecordBatch::new(schema.clone(), batch1_columns).unwrap();
+
+        let batch2_columns: Vec<VectorRef> = vec![
+            Arc::new(UInt32Vector::from_slice(vec![4, 5, 6])),
+            Arc::new(StringVector::from(vec!["David", "Eva", "Frank"])),
+            Arc::new(Float64Vector::from_slice(vec![40.1, 50.2, 60.3])),
+        ];
+        let batch2 = RecordBatch::new(schema.clone(), batch2_columns).unwrap();
+
+        let batch3_columns: Vec<VectorRef> = vec![
+            Arc::new(UInt32Vector::from_slice(vec![7, 8, 9])),
+            Arc::new(StringVector::from(vec!["Grace", "Henry", "Ivy"])),
+            Arc::new(Float64Vector::from_slice(vec![70.4, 80.5, 90.6])),
+        ];
+        let batch3 = RecordBatch::new(schema.clone(), batch3_columns).unwrap();
+
+        // Combine all batches into a RecordBatches collection
+        let recordbatches = RecordBatches::try_new(schema, vec![batch1, batch2, batch3]).unwrap();
+
+        // Test with different compression types
+        let compression_types = vec![
+            CompressionType::Gzip,
+            CompressionType::Bzip2,
+            CompressionType::Xz,
+            CompressionType::Zstd,
+        ];
+
+        for compression_type in compression_types {
+            let format = JsonFormat {
+                compression_type,
+                ..JsonFormat::default()
+            };
+
+            // Create a temporary file path
+            let temp_dir = common_test_util::temp_dir::create_temp_dir("test_compressed_json");
```

Comment:
Nit: temp_dir can be moved out of the loop.

Reply:
You are right.
Done.